src/init/starship.ps1

```diff
@@ -14,9 +14,30 @@ function global:prompt {
     $current_directory = (Convert-Path -LiteralPath $PWD)
     
     # If an external command has not been executed, then the $LASTEXITCODE will be $null.
-    # For the purposes of the prompt, replace this with a 0 exit code so that the prompt
-    # doesn't show the last command as a failure (because it had a non-zero exit code).
-    $lastExitCodeForPrompt = if ($origLastExitCode) { $origLastExitCode } else { 0 }
+    # it isn't populated or modified until you execute a Powershell task that returns an exit
+    # code e.g. On a fresh Powershell console:
+    # > $LASTEXITCODE -eq $null
+    # True
+    # In the same console after running a Powershell cmdlet:
+    # > gci ~/non_existing_folder
+    # ...
+    # > $LASTEXITCODE -eq $null
+    # True
+    # It only gets populated when running an external executable that returns an exit code, e. g.
+    # > bash -c 'exit 5'
+    # > $LASTEXITCODE -eq $null
+    # False
+    # > $LASTEXITCODE
+    # 5
+    # Therefore, for the purpose of the prompt, to avoid coloring the prompt wrongly we should rely
+    # rather on the dollar hook ($?) to check for success/failure of the previous execution.
+    # There's some edge cases like iex where the execution is successful even when the command passed
+    # as parameter returns a failure exit code or fails to execute.
+    # As of Powershell 7 we can leaverage the ternary operator as follows:
+    # $lastExitCodeForPrompt = $origDollarQuestion ? 0 : 1
+    # But for backwards compatibility sake, we can get away with:
+
+    $lastExitCodeForPrompt = @({1}, {0})[$origDollarQuestion]
```

Comment:
```suggestion
    $lastExitCodeForPrompt = 0
    if (-not $origDollarQuestion) {
         $lastExitCodeForPrompt = 1
         if ($origLastExitCode -neq $null) {
             $lastExitCodeForPrompt = $origLastExitCode
         }
    }
```
(I haven't tested this code yet)
I'd prefer something like this that preserves the exit code when it's available.

Reply:
The only issue I see about the extra effort to preserve the exit code is that we can get to this problem:
```
> bash -c 'exit 5'
> $Global:LASTEXITCODE
5
> gci inexistentfolder
Get-ChildItem: Cannot find path '/home/my-user/inexistentfolder' because it does not exist
> $Global:LASTEXITCODE
5
``` 
I would check it myself later, but it would be easier if you can point me to a place in the Rust code where the exit code is relevant for some process, then I would totally agree that it is necessary to find to preserve the exit code and a way to tell apart whether the last command was an internal Powershell (exit code is irrelevant) command or an external one (exit code is relevant), if the exit code is only used in the `starship` command to define the color/symbol of the prompt I would argue that a simple 0 or 1 is enough.

Reply:
The exit code value is needed for displaying it in the [`status`  module](https://starship.rs/config/#status).